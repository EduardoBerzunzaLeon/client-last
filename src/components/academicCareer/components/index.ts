export * from './StudentCard';
export * from './UserCreatorCard';
