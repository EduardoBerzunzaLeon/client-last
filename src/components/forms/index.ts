export * from './fileInput';
export * from './inputs';
export * from './radioButton';
export * from './toggleButton';
