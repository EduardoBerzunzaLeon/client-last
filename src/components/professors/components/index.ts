export * from './ButtonProfessorActive';
export * from './columns';
export * from './course';
export * from './ExcelButtonProfessors';
export * from './Header';
export * from './ProfessorDataForm';
export * from './ProfessorDialog';
