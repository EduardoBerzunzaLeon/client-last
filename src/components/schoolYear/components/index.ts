export * from './PhaseCard';
export * from './FileErrorsTable';
export * from './CurrentSubjectsForm';
export * from './FailedSubjectsForm';
export * from './ValidationPasswordForm';
export * from './SchoolYearStepForm';
