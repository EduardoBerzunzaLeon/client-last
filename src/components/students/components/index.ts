export * from './columns';
export * from './Header';
export * from './StudentDataForm';
export * from './StudentDialog';
export { ProfessorsHistoryDialog } from './professorsHistory';
