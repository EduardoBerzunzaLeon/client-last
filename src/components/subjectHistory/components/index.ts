export * from './SubjectsStudied';
export * from './CurrentSubjects';
export * from './UnstudySubjects';
export * from './columns';
export * from './PhaseDialog';
export * from './Header';
export * from './PhaseDataForm';
