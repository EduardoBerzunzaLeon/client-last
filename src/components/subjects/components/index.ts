export * from './Header';
export * from './columns';
export * from './SubjectDialog';
export * from './SubjectsList';
export * from './SubjectDetailDialog';
