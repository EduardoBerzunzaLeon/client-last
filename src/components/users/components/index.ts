export * from './AdminPasswordForm';
export * from './ButtonUserBlocked';
export * from './columns';
export * from './Header';
export * from './UserDataForm';
export * from './UserDialog';
