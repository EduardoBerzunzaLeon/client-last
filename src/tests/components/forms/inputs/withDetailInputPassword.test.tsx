import { mount } from 'enzyme';
import { Formik } from 'formik';

import { InputTextApp, withDetailInputPassword } from '../../../../components/forms';

describe('<withDetailInputPassword />', () => {
  test('should match to snapshot and input password must have footerinputpassword props', async () => {
    const InputPassword = withDetailInputPassword(InputTextApp);

    const wrapper = mount(
      <Formik
        initialValues={{
          password: '',
        }}
        onSubmit={() => {}}
      >
        {() => (
          <InputPassword
            label="Contraseña"
            name="password"
            className="w-full"
          />

        )}
      </Formik>,
    );

    // ? is necessary in change event formik
    // https://medium.com/@jorgeortega/react-forms-with-formik-and-unit-testing-with-react-testing-library-e2dda1a899db
    // const password = container.querySelector('input[name="password"]');
    // if (password) {
    //   await waitFor(() => {
    //     fireEvent.change(password, {
    //       target: {
    //         value: 'mockpassword',
    //       },
    //     });
    //   });
    // }

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Password').props()).toEqual(expect.objectContaining({
      promptLabel: 'Ingresa una contraseña',
      weakLabel: 'Débil',
      mediumLabel: 'Moderada',
      strongLabel: 'Difícil',
      type: 'password',
    }));
  });
});
