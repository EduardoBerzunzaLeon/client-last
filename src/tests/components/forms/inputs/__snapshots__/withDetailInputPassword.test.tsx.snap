// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withDetailInputPassword /> should match to snapshot and input password must have footerinputpassword props 1`] = `
<Formik
  initialValues={
    Object {
      "password": "",
    }
  }
  onSubmit={[Function]}
>
  <Component
    className="w-full"
    label="Contraseña"
    name="password"
  >
    <InputTextApp
      className="w-full"
      footer={[Function]}
      icon=""
      id=""
      keyfilter={/\\[\\^\\\\s\\]/}
      label="Contraseña"
      mediumLabel="Moderada"
      name="password"
      placeholder=""
      promptLabel="Ingresa una contraseña"
      strongLabel="Difícil"
      toggleMask={true}
      type="password"
      weakLabel="Débil"
    >
      <span
        className="p-float-label p-input-icon-right w-full"
      >
        <i
          className=""
        />
        <Password
          appendTo={null}
          autoComplete="off"
          className="w-full"
          content={null}
          feedback={true}
          footer={[Function]}
          header={null}
          icon=""
          id=""
          inputClassName={null}
          inputId={null}
          inputRef={null}
          inputStyle={null}
          keyfilter={/\\[\\^\\\\s\\]/}
          mediumLabel="Moderada"
          mediumRegex="^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onHide={null}
          onInput={null}
          onShow={null}
          panelClassName={null}
          panelStyle={null}
          placeholder=""
          promptLabel="Ingresa una contraseña"
          strongLabel="Difícil"
          strongRegex="^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})"
          style={null}
          toggleMask={true}
          tooltip={null}
          tooltipOptions={null}
          transitionOptions={null}
          type="password"
          value=""
          weakLabel="Débil"
        >
          <div
            className="p-password p-component p-inputwrapper p-input-icon-right w-full"
            id=""
            style={null}
          >
            <ForwardRef
              autoComplete="off"
              className="p-password-input"
              id={null}
              keyfilter={/\\[\\^\\\\s\\]/}
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyUp={[Function]}
              placeholder=""
              style={null}
              type="password"
              value=""
            >
              <InputTextComponent
                autoComplete="off"
                className="p-password-input"
                forwardRef={
                  Object {
                    "current": <input
                      autocomplete="off"
                      class="p-inputtext p-component p-password-input"
                      name="password"
                      placeholder=""
                      type="password"
                      value=""
                    />,
                  }
                }
                id={null}
                keyfilter={/\\[\\^\\\\s\\]/}
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyPress={null}
                onKeyUp={[Function]}
                placeholder=""
                style={null}
                tooltip={null}
                tooltipOptions={null}
                type="password"
                validateOnly={false}
                value=""
              >
                <input
                  autoComplete="off"
                  className="p-inputtext p-component p-password-input"
                  id={null}
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder=""
                  style={null}
                  type="password"
                  value=""
                />
              </InputTextComponent>
            </ForwardRef>
            <i
              className="pi pi-eye"
              onClick={[Function]}
            />
            <Portal
              appendTo={null}
              element={
                <CSSTransition
                  classNames="p-connected-overlay"
                  in={false}
                  nodeRef={
                    Object {
                      "current": null,
                    }
                  }
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  options={null}
                  timeout={
                    Object {
                      "enter": 120,
                      "exit": 100,
                    }
                  }
                  unmountOnExit={true}
                >
                  <div
                    className="p-password-panel p-component"
                    onClick={[Function]}
                    style={null}
                  >
                    <React.Fragment>
                      <div
                        className="p-password-meter"
                      >
                        <div
                          className="p-password-strength "
                          style={
                            Object {
                              "width": "0%",
                            }
                          }
                        />
                      </div>
                      <div
                        className="p-password-info"
                      >
                        Ingresa una contraseña
                      </div>
                    </React.Fragment>
                    <React.Fragment>
                      <Divider
                        align={null}
                        className={null}
                        layout="horizontal"
                        style={null}
                        type="solid"
                      />
                      <p
                        className="p-mt-2"
                      >
                        Sugerencias
                      </p>
                      <ul
                        className="p-pl-2 p-ml-2 p-mt-0"
                        style={
                          Object {
                            "lineHeight": "1.5",
                          }
                        }
                      >
                        <li>
                          Al menos una minúscula
                        </li>
                        <li>
                          Al menos una mayúscula
                        </li>
                        <li>
                          Al menos un númerico
                        </li>
                        <li>
                          Mínimo 8 carácteres
                        </li>
                      </ul>
                    </React.Fragment>
                  </div>
                </CSSTransition>
              }
              onMounted={null}
              onUnmounted={null}
              visible={false}
            >
              <Portal
                containerInfo={
                  <body>
                    <script
                      id="facebook-jssdk"
                    />
                  </body>
                }
              >
                <CSSTransition
                  classNames="p-connected-overlay"
                  in={false}
                  nodeRef={
                    Object {
                      "current": null,
                    }
                  }
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  options={null}
                  timeout={
                    Object {
                      "enter": 120,
                      "exit": 100,
                    }
                  }
                  unmountOnExit={true}
                >
                  <CSSTransition
                    classNames="p-connected-overlay"
                    in={false}
                    nodeRef={
                      Object {
                        "current": null,
                      }
                    }
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={
                      Object {
                        "enter": 120,
                        "exit": 100,
                      }
                    }
                    unmountOnExit={true}
                  >
                    <Transition
                      appear={false}
                      enter={true}
                      exit={true}
                      in={false}
                      mountOnEnter={false}
                      nodeRef={
                        Object {
                          "current": null,
                        }
                      }
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={
                        Object {
                          "enter": 120,
                          "exit": 100,
                        }
                      }
                      unmountOnExit={true}
                    />
                  </CSSTransition>
                </CSSTransition>
              </Portal>
            </Portal>
          </div>
        </Password>
        <label
          className=""
          htmlFor="password"
        >
          Contraseña
        </label>
      </span>
      <FormikConnect(ErrorMessageImpl)
        name="password"
      >
        <ErrorMessageImpl
          formik={
            Object {
              "dirty": false,
              "errors": Object {},
              "getFieldHelpers": [Function],
              "getFieldMeta": [Function],
              "getFieldProps": [Function],
              "handleBlur": [Function],
              "handleChange": [Function],
              "handleReset": [Function],
              "handleSubmit": [Function],
              "initialErrors": Object {},
              "initialStatus": undefined,
              "initialTouched": Object {},
              "initialValues": Object {
                "password": "",
              },
              "isSubmitting": false,
              "isValid": true,
              "isValidating": false,
              "registerField": [Function],
              "resetForm": [Function],
              "setErrors": [Function],
              "setFieldError": [Function],
              "setFieldTouched": [Function],
              "setFieldValue": [Function],
              "setFormikState": [Function],
              "setStatus": [Function],
              "setSubmitting": [Function],
              "setTouched": [Function],
              "setValues": [Function],
              "status": undefined,
              "submitCount": 0,
              "submitForm": [Function],
              "touched": Object {},
              "unregisterField": [Function],
              "validateField": [Function],
              "validateForm": [Function],
              "validateOnBlur": true,
              "validateOnChange": true,
              "validateOnMount": false,
              "values": Object {
                "password": "",
              },
            }
          }
          name="password"
        />
      </FormikConnect(ErrorMessageImpl)>
    </InputTextApp>
  </Component>
</Formik>
`;
